File: AdMobMediation_v8/ViewController/AdItemViewController.m
//
//  AdItemViewController.m
//  GoogleMediation
//
//  Created by JustinTsou on 2021/4/23.
//

#import "AdItemViewController.h"
#import "NativeAdViewController.h"
#import "SuprAdViewController.h"
#import "BannerViewController.h"

typedef NS_ENUM(NSInteger, AdEnum) {
    NativeAd = 0,
    SuprAd = 1,
    SuprAdBanner = 2
};

@interface AdItemViewController ()<UITableViewDataSource, UITableViewDelegate>
{
    NSArray *_adItem;
}
@property (weak, nonatomic) IBOutlet UITableView *adItemTableView;
@property AdEnum adEnum;

@end

@implementation AdItemViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _adItem = [[NSArray alloc]initWithObjects:@"Native Ad",@"Supr Ad",@"Banner Ad", nil];
    
    [self setupTableVie];
}

#pragma mark : Setup TableView

- (void)setupTableVie {
    self.adItemTableView.dataSource = self;
    self.adItemTableView.delegate = self;
    
    [self.adItemTableView registerClass:UITableViewCell.class forCellReuseIdentifier:@"Cell"];
}


#pragma mark - UITableViewDataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _adItem.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {

    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    cell.textLabel.text = _adItem[indexPath.row];
    return  cell;
}


#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    if (indexPath.row == NativeAd) {
        NativeAdViewController *nativeAdViewController = [[NativeAdViewController alloc]init];
        [self.navigationController pushViewController:nativeAdViewController animated:YES];
    }else if (indexPath.row == SuprAd) {
        SuprAdViewController *suprAdViewController = [[SuprAdViewController alloc]init];
        [self.navigationController pushViewController:suprAdViewController animated:YES];
    }if (indexPath.row == SuprAdBanner) {
        BannerViewController *bannerViewController = [[BannerViewController alloc]init];
        [self.navigationController pushViewController:bannerViewController animated:YES];
    }
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 50;
}

@end

File: AdMobMediation_v8_Swift/ViewController/AdItemViewController.swift
//
//  AdItemViewController.swift
//  AotterGoogleMediationAdSwift
//
//  Created by JustinTsou on 2021/4/23.
//

import UIKit

enum AdEnum:Int {
    case NativeAd
    case SuprAd
    case SuprAdBanner
    
    static func getReviewCellType() -> [AdEnum] {
        return [.NativeAd,.SuprAd,SuprAdBanner]
    }
}

class AdItemViewController: UIViewController {
    
    @IBOutlet weak var adItemTableView: UITableView!
    
    private let adItem = ["Native Ad","Supr Ad","Banner Ad"];

    override func viewDidLoad() {
        super.viewDidLoad()

        self.setupTableView()
    }

}

// MARK: Setup UI

extension AdItemViewController {
    
    private func setupTableView() {
        adItemTableView.delegate = self
        adItemTableView.dataSource = self
        adItemTableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
    }
}

// MARK: UITableViewDataSource

extension AdItemViewController: UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return adItem.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        cell.textLabel?.text = adItem[indexPath.row]
        return cell
    }
}

extension AdItemViewController: UITableViewDelegate {
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let row = AdEnum.getReviewCellType()[indexPath.row]
        
        switch row {
        case .NativeAd:
            let nativeAdViewController = NativeAdViewController(nibName: "NativeAdViewController", bundle: nil)
            self.navigationController?.pushViewController(nativeAdViewController, animated: true)
            break
        case .SuprAd:
            let suprAdViewController = SuprAdViewController(nibName: "SuprAdViewController", bundle: nil)
            self.navigationController?.pushViewController(suprAdViewController, animated: true)
            break
        case .SuprAdBanner:
            let bannerViewController = BannerViewController(nibName: "BannerViewController", bundle: nil)
            self.navigationController?.pushViewController(bannerViewController, animated: true)
            break
        }
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 50
    }
}
